logstash-core/src/main/java/org/logstash/instrument/witness/QueueWitness.java

```diff
@@ -0,0 +1,60 @@
+package org.logstash.instrument.witness;
+
+import org.logstash.instrument.metrics.gauge.TextGauge;
+
+/**
+ * Witness for the queue.
+ */
+final public class QueueWitness {
+
+    private final TextGauge type;
+    private final Snitch snitch;
+
+    /**
+     * Constructor.
+     */
+    public QueueWitness() {
+        type = new TextGauge("type");
+        snitch = new Snitch(this);
+    }
+
+    /**
+     * Get a reference to associated snitch to get discrete metric values.
+     *
+     * @return the associate {@link Snitch}
+     */
+    public Snitch snitch() {
+        return snitch;
+    }
+
+    /**
+     * Sets the type of the queue.
+     *
+     * @param type The type of the queue.
+     */
+    public void type(String type) {
```

Comment:
+1 to anything that makes it more readable.  

For this iteration I was just matching like for like the existing name space. I will log a new issue to revisit the naming in whole. Mind if we wait till after a few more reviews to start those changes (it makes the mental diffing with namespaces easier) ?